xarray/backends/zarr.py

```diff
@@ -34,11 +35,97 @@
     from xarray.backends.common import AbstractDataStore
     from xarray.core.dataset import Dataset
 
-
 # need some special secret attributes to tell us the dimensions
 DIMENSION_KEY = "_ARRAY_DIMENSIONS"
 
 
+def initialize_zarr(
+    store,
+    ds: Dataset,
+    *,
+    region_dims: Iterable[Hashable] | None = None,
+    mode: Literal["w", "w-"] = "w-",
+    **kwargs,
+) -> Dataset:
+    """
+    Initialize a Zarr store with metadata.
+
+    This function initializes a Zarr store with metadata that describes the entire datasets.
+    If ``region_dims`` is specified, it will also
+    1. Write variables that don't contain any of ``region_dims``, and
+    2. Return a dataset with variables that do contain one or more of ``region_dims``.
+       This dataset can be used for region writes in parallel.
+
+    Parameters
+    ----------
+    store : MutableMapping or str
+        Zarr store to write to.
+    ds : Dataset
+        Dataset to write.
+    region_dims : Iterable[Hashable], optional
+        An iterable of dimension names that will be passed to the ``region``
+        kwarg of ``to_zarr`` later.
+    mode : {'w', 'w-'}
+        Write mode for initializing the store.
+
+    Returns
+    -------
+    Dataset
+        Dataset containing variables with one or more ``region_dims``
+        dimensions. Use this for writing to the store in parallel later.
+
+    Raises
+    ------
+    ValueError
+
+    """
+
+    if "compute" in kwargs:
+        raise ValueError("The ``compute`` kwarg is not supported in `initialize_zarr`.")
+
+    if not ds.chunks:
+        raise ValueError("This function should be used with chunked Datasets.")
+
+    if mode not in ["w", "w-"]:
+        raise ValueError(
+            f"Only mode='w' or mode='w-' is allowed for initialize_zarr. Received mode={mode!r}"
+        )
+
+    # TODO: what should we do here.
+    # compute=False only skips dask variables.
+    # - We could reaplce all dask variables with zeros_like
+    # - and then write all other variables eagerly.
+    # Right now we do two writes for eager variables
+    template = zeros_like(ds)
+    template.to_zarr(store, mode=mode, **kwargs, compute=False)
+
```

Comment:
☝🏾 This is important. What do we mean by initialization? 
1. All non-dask arrays will be written to disk. & Metadata for all dask arrays will be written to disk.
2. Alternatively, it could be all metadata for all variables will be written to disk. Reading that initialized store will get you fill values.

Reply:
As a concrete example we could do soemthing like this from pangeo-forge:
https://github.com/pangeo-forge/pangeo-forge-recipes/blob/5367f6363be91dd4826f3e5f21ffea61b60994cb/pangeo_forge_recipes/aggregation.py#L205

(thanks @rabernat)

Reply:
I was originally thinking that it would be the former. It's at least important to write indexes, not just their metadata...

Reply:
If you don't write the dimension coordinates, the dataset may be unreadable by xarray. The fill values could decode to invalid results. This happens especially with time coordinates.